staging/src/k8s.io/apiserver/pkg/storage/cacher/watch_cache.go

```diff
@@ -388,20 +388,22 @@ func (w *watchCache) UpdateResourceVersion(resourceVersion string) {
 		return
 	}
 
-	w.Lock()
-	defer w.Unlock()
-	w.resourceVersion = rv
-
-	// Don't dispatch bookmarks coming from the storage layer.
-	// They can be very frequent (even to the level of subseconds)
-	// to allow efficient watch resumption on kube-apiserver restarts,
-	// and propagating them down may overload the whole system.
-	//
-	// TODO: If at some point we decide the performance and scalability
-	// footprint is acceptable, this is the place to hook them in.
-	// However, we then need to check if this was called as a result
-	// of a bookmark event or regular Add/Update/Delete operation by
-	// checking if resourceVersion here has changed.
+	go func() {
+		w.Lock()
+		defer w.Unlock()
+		w.resourceVersion = rv
+	}()
+
+	// Avoid calling event handler under lock.
+	// This is safe as long as there is at most one call to processEvent in flight
```

Comment:
which processEvent? The one in this file? The relation between that function and this one isn't obvious, so it's really hard to be sure that that condition holds.

Reply:
Copy-paste issue - will update.